### PR TITLE
Add amsmath and admonition container parsing

### PR DESCRIPTION
**Pull request by chrisjsewell**

### CODE REVIEW

**Comment by chrisjsewell** [comment]
Hows this @choldgraf @jstac @mmcky @phaustin 😄 
See https://myst-parser--201.org.readthedocs.build/en/201/using/syntax.html#direct-latex-math-optional

**Comment by chrisjsewell** [comment]
Plus documented the other configuration for normal math file:///Users/chrisjsewell/Documents/GitHub/myst_parser_ebp/docs/_build/html/using/syntax.html#math-shortcuts

**Comment by phaustin** [comment]
looks good to me -- I do think it's worth the extra space to add one more textbox to https://myst-parser--201.org.readthedocs.build/en/201/using/syntax.html#direct-latex-math-optional below the note, showing a labeled version of (3) --  that would drive home the point that any equation can be labeled, as long as you add the `$$`.

**Comment by jstac** [comment]
Sweet :-)  Thanks @chrisjsewell 

I've really been missing `multline`.

File: myst_parser/__init__.py [hunk 1]
```diff
@@ -5,10 +5,40 @@ def setup(app):
...
+
+    allowed_delimiters = ["brackets", "kramdown", "dollars", "julia"]
+    if config.myst_math_delimiters not in allowed_delimiters:
+        logger.error(
+            "myst_math_delimiters config option not an allowed name: "
+            + f"{allowed_delimiters}"
+        )
+
+    if not isinstance(config.myst_amsmath_enable, bool):
+        logger.error("myst_disable_syntax config option not of type boolean")
```
Comment (chrisjsewell) [inline myst_parser/__init__.py:44]:
```suggestion
        logger.error("myst_amsmath_enable config option not of type boolean")
```

**Comment by chrisjsewell** [comment]
prodding 👉 @choldgraf to review this 😉

**Pull request merged**